ui/app/contexts/metametrics.new.js

```diff
@@ -128,72 +127,53 @@ export function MetaMetricsProvider({ children }) {
    */
   useEffect(() => {
     const environmentType = getEnvironmentType()
-    if (
-      (participateInMetaMetrics === null &&
-        location.pathname.startsWith('/initialize')) ||
-      participateInMetaMetrics
-    ) {
-      // Events that happen during initialization before the user opts into
-      // MetaMetrics will be anonymous
-      const idTrait = metaMetricsId ? 'userId' : 'anonymousId'
-      const idValue = metaMetricsId ?? METAMETRICS_ANONYMOUS_ID
-      const match = matchPath(location.pathname, {
-        path: PATHS_TO_CHECK,
-        exact: true,
-        strict: true,
-      })
-      // Start by checking for a missing match route. If this falls through to
-      // the else if, then we know we have a matched route for tracking.
-      if (!match) {
+    // Events that happen during initialization before the user opts into MetaMetrics will be anonymous
+    const match = matchPath(location.pathname, {
+      path: PATHS_TO_CHECK,
+      exact: true,
+      strict: true,
+    })
+    // Start by checking for a missing match route. If this falls through to the else if, then we know we
+    // have a matched route for tracking.
+    if (!match) {
+      // We have more specific pages for each type of transaction confirmation
+      // The user lands on /confirm-transaction first, then is redirected based on
+      // the contents of state.
+      if (location.pathname !== '/confirm-transaction') {
```

Comment:
Wasn't this case removed in a previous PR? I thought `confirm-transaction` was added to the list of paths. Maybe this was reintroduced during a rebase accidentally?

Reply:
Thanks, restored the prior fix.